pkg/prometheus/resource_selector.go

```diff
@@ -1441,3 +1445,33 @@ func (rs *ResourceSelector) validateIonosSDConfigs(ctx context.Context, sc *moni
 	}
 	return nil
 }
+
+func (rs *ResourceSelector) validateStackitSDConfigs(ctx context.Context, sc *monitoringv1alpha1.ScrapeConfig) error {
+	if len(sc.Spec.StackitSDConfigs) == 0 {
```

Comment:
I guess we don't need this check?

Reply:
sorry @slashpai my bad you meant the` len check`, removed, pls review

Reply:
I guess removing this broke the tests, at this point to exactly sure why, will look into it